lib/puppet/indirector/request.rb

```diff
@@ -250,8 +250,6 @@ def set_uri_key(key)
       return
     end
 
-    env, indirector, @key = URI.unescape(uri.path.sub(/^\//, '')).split('/',3)
-    @key ||= ''
-    self.environment = env unless env == ''
+    @key = key
```

Comment:
For prosperity, adding a note about what I was saying at PR triage. Currently, `Dipper` calls `Puppet::FileBucket::File.indirector.head` specifying a `file_bucket_path` of `https://hwvft1j1a4531gm.delivery.puppetlabs.net:8140/production/file_bucket_file/md5/d41d8cd98f00b204e9800998ecf8427e//Users/josh/work/puppet/empty`, and `Request` has special logic, rewriting the key since it's neither a puppet nor file based scheme.

The more "standard" spooky-action-at-a-distance approach would be for the `filebucket` application to configure the termini based on whether a local or remote file bucket is being used (and `inspect` application too). For example, the newer face-based `file` download & store actions do this already.

Then the Dipper can just call `Puppet::FileBucket::File.indirector.head("md5/d41d8cd...")`, and puppet should route the request to the current terminus. We might then be able to able to move forward with the change to `Request` in this PR, and perhaps simplify/remove `lib/puppet/indirector/file_bucket_file/selector.rb`.